open_city_profile/graphene.py

```diff
@@ -178,3 +181,44 @@ def __init_subclass_with_meta__(
             _meta=_meta,
             **options,
         )
+
+
+class AllowedDataFieldsMiddleware:
+
+    def _is_profile_query(self, info):
+        return any(
+            selection.name.value in ("myProfile", "profile", "profiles")
+            for selection in info.operation.selection_set.selections
+        )
+
+    def resolve(self, next, root, info, **kwargs):
+        try:
+            is_profile_query = self._is_profile_query(info)
+            node = info.parent_type.graphene_type
+        except AttributeError:
+            is_profile_query = False
+            node = None
+
+        if is_profile_query and getattr(node, "check_allowed_data_fields", False):
+            field_name = to_snake_case(getattr(info, "field_name", ""))
+
+            if not getattr(info.context, "service", False):
+                if settings.ENABLE_ALLOWED_DATA_FIELDS_RESTRICTION:
+                    raise ServiceNotIdentifiedError("Service not identified")
+
+                logging.warning(
+                    "Allowed data field exception would occur: Service not identified. Field name: %s",
+                    field_name,
+                )
+
+            if not node.is_field_allowed_for_service(field_name, info.context.service):
+                if settings.ENABLE_ALLOWED_DATA_FIELDS_RESTRICTION:
+                    raise FieldNotAllowedError("Field is not allowed for service.")
+
+                logging.warning(
+                    "Allowed data field exception would occur. Field (%s) is not allowed for service %s. Field",
```

Comment:
There might be an extra `Field` at the end of this warning message.

Reply:
Nice catch 🎣